python_dwd/request_dwd.py

```diff
@@ -0,0 +1,226 @@
+
+        return station_id
+
+    @staticmethod
+    def clean_string(string):
+        return string.strip().lower().split("_")
+
+    @staticmethod
+    def find_any_one_word_from_wordlist(string_list, word_list):
+        return all([any([any([(word in string) if not word.isdigit() else word == string
+                        for word in wl]) for string in string_list])
+                   for wl in word_list])
+
+
+class DWDRequest:
+    def __init__(self,
+                 station_id: List[Union[int, str]],
+                 parameter: Parameter,
+                 period_type: PeriodType,
+                 time_resolution: Union[None, str, list, TimeResolution] = None,
+                 start_date: Union[None, str, Timestamp] = None,
+                 end_date: Union[None, str, Timestamp] = None):
+        if not isinstance(station_id, list):
+            try:
+                station_id = [station_id]
+            except ValueError:
+                raise TypeError("Error: station_id is not of type list")
+
+        if not isinstance(parameter, (str, Parameter)):
+            raise TypeError("Error: parameter is not an instance of str or the Parameter enumeration.")
+
+        if not isinstance(period_type, (str, PeriodType)):
+            raise TypeError("Error: period_type is not an instance of str or the PeriodType enumeration.")
+
+        if not (time_resolution or (start_date and end_date)):
+            raise ValueError(
+                "Define either a time_resolution or both the start_ and end_date and leave the other one empty!")
+        if time_resolution:
+            if not isinstance(time_resolution, (str, TimeResolution)):
+                raise TypeError("Error: time_resolution is not an instance of str or the TimeResolution enumeration.")
+
+            if not isinstance(time_resolution, list):
+                try:
+                    time_resolution = [time_resolution]
+                except ValueError:
+                    raise TypeError("Error: time_resolution can't be converted to a list.")
+
+        if start_date and end_date:
+            start_date = FuzzyExtractor("start_date", start_date).extract_parameter_from_value()
+            end_date = FuzzyExtractor("end_date", end_date).extract_parameter_from_value()
+
+            if not start_date <= end_date:
+                raise ValueError("Error: end_date must at least be start_date or later!")
+
+            time_resolution = [*TimeResolution]
+
+        self.station_id = FuzzyExtractor("station_id", station_id).extract_parameter_from_value()
+        self.parameter = FuzzyExtractor("parameter", parameter).extract_parameter_from_value()
+        self.period_type = FuzzyExtractor("period_type", period_type).extract_parameter_from_value()
+        self.time_resolution = [FuzzyExtractor("time_resolution", time_res).extract_parameter_from_value()
+                                for time_res in time_resolution]
```

Comment:
Why do you build a class FuzzyExtractor?! Looks more like a function?

Reply:
It's for overview purposes as it uses three different methods depending on which argument is parsed.